### ISSUE TITLE
[BUG]: Error during DeepFace analysis: The layer sequential_1 has never been called and thus has no defined input.

### ISSUE THREAD

**Issue by tora-golia** [serengil/deepface#1380]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [X] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

0.0.93

### Python version

3.12.4

### Operating System

macOS 14.6.1

### Dependencies

absl-py==2.1.0
annotated-types==0.7.0
anyio==4.4.0
astor==0.8.1
astunparse==1.6.3
beautifulsoup4==4.12.3
blinker==1.8.2
certifi==2024.8.30
charset-normalizer==3.3.2
click==8.1.7
contourpy==1.3.0
cycler==0.12.1
Cython==3.0.11
decorator==4.4.2
deepface==0.0.93
distro==1.9.0
dlib==19.24.6
face-recognition==1.3.0
face_recognition_models==0.3.0
facenet-pytorch==2.6.0
fastapi==0.115.0
fer==22.5.1
ffmpeg==1.4
filelock==3.16.1
fire==0.6.0
Flask==3.0.3
Flask-Cors==5.0.0
flatbuffers==24.3.25
fonttools==4.53.1
fsspec==2024.9.0
gast==0.6.0
gdown==5.2.0
google-pasta==0.2.0
grpcio==1.66.1
gunicorn==23.0.0
h11==0.14.0
h5py==3.11.0
httpcore==1.0.5
httpx==0.27.2
idna==3.10
imageio==2.35.1
imageio-ffmpeg==0.5.1
imgaug==0.4.0
imutils==0.5.4
itsdangerous==2.2.0
Jinja2==3.1.4
jiter==0.6.1
joblib==1.4.2
keras==3.5.0
kiwisolver==1.4.7
lazy_loader==0.4
libclang==18.1.1
lmdb==1.5.1
lxml==5.3.0
lz4==4.3.3
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.9.2
mdurl==0.1.2
ml-dtypes==0.4.1
moviepy==1.0.3
mpmath==1.3.0
mtcnn==1.0.0
namex==0.0.8
networkx==3.3
numpy==1.26.4
openai==1.51.2
opencv-contrib-python==4.10.0.84
opencv-python==4.10.0.84
opt-einsum==3.3.0
optree==0.12.1
packaging==24.1
paddleocr==2.8.1
paddlepaddle==2.6.2
pandas==2.2.2
pillow==10.2.0
proglog==0.1.10
protobuf==4.25.4
pyclipper==1.3.0.post5
pydantic==2.9.2
pydantic_core==2.23.4
Pygments==2.18.0
pyparsing==3.1.4
PySocks==1.7.1
pytesseract==0.3.13
python-dateutil==2.9.0.post0
python-docx==1.1.2
python-multipart==0.0.12
pytz==2024.2
PyYAML==6.0.2
rapidfuzz==3.9.7
requests==2.32.3
retina-face==0.0.17
rich==13.8.1
scikit-image==0.24.0
scipy==1.14.1
setuptools==75.1.0
shapely==2.0.6
six==1.16.0
sniffio==1.3.1
soupsieve==2.6
starlette==0.38.5
sympy==1.13.2
tensorboard==2.18.0
tensorboard-data-server==0.7.2
tensorflow==2.18.0
termcolor==2.4.0
tf_keras==2.18.0
tifffile==2024.8.30
torch==2.2.2
torchvision==0.17.2
tqdm==4.66.5
typing_extensions==4.12.2
tzdata==2024.1
urllib3==2.2.3
uvicorn==0.30.6
Werkzeug==3.0.4
wheel==0.44.0
wrapt==1.16.0

### Reproducible example

```Python
from deepface import DeepFace
from pydantic import BaseModel
from fastapi import APIRouter, HTTPException
router = APIRouter()

class DeepfaceDetectionRequest(BaseModel):
    image_url: str

@router.post("/deepface-detection")
async def deepface_detection(request: DeepfaceDetectionRequest):
    print(f"Received deepface detection request for URL: {request.image_url}")

    try:
        result = DeepFace.analyze(request.image_url)

    except Exception as e:
        print(f"Error during DeepFace analysis: {e}")
        raise HTTPException(status_code=400, detail="DeepFace detection failed")

    return result
```

### Relevant Log Output

Received deepface detection request for URL: https://i.ytimg.com/vi/07prS6_oaXc/maxresdefault.jpg
Action: age:  25%|██▌       | 1/4 [00:00<00:01,  1.97it/s]
Error during DeepFace analysis: The layer sequential_1 has never been called and thus has no defined input.
INFO:     127.0.0.1:65419 - "POST /deepface-detection HTTP/1.1" 400 Bad Request

### Additional Info

When I change the code to `result = DeepFace.analyze(request.image_url, ['emotion'])`, it runs successfully.

**Comment by serengil** [comment]
This should be fixed in the latest source code - https://github.com/serengil/deepface/pull/1357/files#diff-c30a7465dd4a31fa7afacb2f7f1004ce72de6152949bc92122689698cf151f75L154

But this is not pushed to pip yet, would you please try to use source code?
